c++/src/kj/compat/gzip.c++

```diff
@@ -232,9 +232,9 @@ Promise<size_t> GzipAsyncInputStream::readImpl(
     }
   } else {
     if (ctx.msg == nullptr) {
-      KJ_FAIL_REQUIRE("gzip decompression failed", inflateResult);
+      return KJ_EXCEPTION(FAILED, "gzip decompression failed", inflateResult);
```

Comment:
The usual convention in KJ is that an async function is free to throw exception either synchronously or asynchronously. If the caller wants to force all exceptions to be async, it is their responsibility to wrap in `kj::evalNow()` or something like that.

So the change here doesn't really make a difference unless the caller is buggy.

What motivated this?